### PR TITLE
Try to firm up flaky RPC tests

### PR DESCRIPTION
**Pull request by pipermerriam**
## What was wrong?

There was some fixture cleanup in the RPC tests and a subtle mis-use of the `trio.Condition` object used in RFN

## How was it fixed?

Removed the redundant fixture code

Moved the acqusition of the condition lock to the outermost layer to avoid race conditions where the condition is triggered between the time when the lock is released and acquired.

#### Cute Animal Picture

adelie-penguin-jumping-ocean ngsversion 1396530997321 adapt 1900 1

### CODE REVIEW

File: ddht/v5_1/network.py [hunk 1]
```diff
@@ -230,11 +230,11 @@ async def worker(
...
-            while True:
-                # this `fail_after` is a failsafe to prevent deadlock situations
-                # which are possible with `Condition` objects.
-                with trio.fail_after(60):
-                    async with condition:
+            async with condition:
+                while True:
+                    # this `fail_after` is a failsafe to prevent deadlock situations
+                    # which are possible with `Condition` objects.
+                    with trio.fail_after(60):
```
Comment (pipermerriam) [inline ddht/v5_1/network.py:237]: I've been just kicking the CI tests over and over to re-run them to try and find the failures.  So far this PR hasn't had a single timeout.  Not conclusive at all but... I think this is where the flakyness was coming from.

TLDR `Condition` objects seem to be the right tool for certain concurrency situations but they are also easy to use incorrectly (and potentially end up with deadlocks).
